### PR TITLE
added dates to specification link

### LINKED ISSUE #123: Docs request: adding dates to Specification Links
JSON schema 2019-09 and 2020-12 use [CalVer](https://calver.org/), so their release date is self explanatory.

It would be useful if the [Specification Links](https://json-schema.org/specification-links.html) page documented release dates for versions (e.g. draft 6, draft 7) before JSON schema started using CalVer.

Something like in the [Draft 7 section here](https://json-schema.org/specification-links.html#draft-7), just adding a bullet for the release date

### PR DESCRIPTION
**Pull request by praveen-rikhari**
**GitHub Issue:** #123
Resolves #123 

**Summary**:
- added published/release dates for each versions (e.g. draft 6, draft 7) 

**Files Changes**
- `pages/specification-links.md`

**Before**
Can see on website [here](https://json-schema.org/specification-links#2020-12)

**After**
[screen-recording.webm](https://github.com/json-schema-org/website/assets/84331681/c15e08ce-aa80-4540-871b-a9e39b397801)

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

Yes/No
No

### CODE REVIEW

**Comment by praveen-rikhari** [comment]
@benjagm Please review

File: pages/specification-links.md [hunk 1]
```diff
@@ -399,24 +404,28 @@ _These were updated without changing functionality or meta-schemas due to a few
  - JSON Reference: [draft-pbryan-zyp-json-ref-03](https://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03) ([changes](https://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03#appendix-A))
  * [JSON Schema meta-schema](../../draft-04/schema)
  * [JSON Hyper-Schema meta-schema](../../draft-04/hyper-schema)
+ - Published: 2013
```
Comment (benjagm) [inline pages/specification-links.md:407]:
```suggestion
 - Published: 31-January-2013
```

**Review by benjagm — CHANGES_REQUESTED** [review]
Just found something in Draft 4.

Great work @praveen-rikhari !!

File: pages/specification-links.md (hunk 1 above)
Comment (praveen-rikhari) [reply pages/specification-links.md:407]: @benjagm Thanks for reviewing
okk I'll fix this

**Comment by praveen-rikhari** [comment]
@benjagm I deleted that #388 and merged the requested changes on to this PR (#386) 
Now you can leave it for review

**Review by benjagm — APPROVED** [review]

**Pull request merged**